src/main/java/net/minecraftforge/event/ForgeEventFactory.java

```diff
@@ -81,6 +83,18 @@
 public class ForgeEventFactory
 {
 
+    public static String onClientChat(EntityPlayerSP sender, String message){
+        ClientChatEvent event = new ClientChatEvent(sender, message);
+        MinecraftForge.EVENT_BUS.post(event);
+        String newmsg = event.getMessage();
+        if(event.isCanceled()){
+            newmsg = null;
+        }else if(newmsg.isEmpty()){
```

Comment:
This could be an or instead, seems a bit wasteful, also, might want to check if newmsg is null first.